scripts/sentry-cli.sh

```diff
@@ -0,0 +1,82 @@
+#!/bin/bash
+# This code is mostly the output of calling `sentry-cli bash-hook`
+# This is used by direnv's execution path since the default behaviour of
+# `sentry-cli bash hook` does not work with it. For instance,
+# 1) We should not trap EXIT for direnv needs that untrapped
+# 2) There's a bug on SENTRY_LOG_FILE that is not fully written to disk before reporting
+set -e
+
+_SENTRY_TRACEBACK_FILE=$(mktemp -t sentry-direnv-envrc.traceback)
+_SENTRY_LOG_FILE=$(mktemp -t sentry-direnv-envrc.out)
+
+# Wait for file to be written to disk
+_wait_for_file() {
+  local filesize
+  for i in {0..10}
+  do
+    filesize=$(du -k "$1" | cut -f1)
+    if [[ $filesize -gt 0 ]]; then
+      break
+    else
+      sleep 0.1
+    fi
+  done
+}
+
+_sentry_err_trap() {
+  local _exit_code="$?"
+  local _command="${BASH_COMMAND:-unknown}"
+  if [ "x$1" != x ]; then
+    _command="$1"
+  fi
+  if [ "x$2" != x ]; then
+    _exit_code="$2"
+  fi
+  _sentry_traceback 1
+  echo "@command:${_command}" >> "$_SENTRY_TRACEBACK_FILE"
+  echo "@exit_code:${_exit_code}" >> "$_SENTRY_TRACEBACK_FILE"
+
+  : >> "$_SENTRY_LOG_FILE"
+  # This line is differing from upstream _sentry_err_trap
+  _wait_for_file "$_SENTRY_LOG_FILE"
+
+  # The specified DSN reports to `sentry-dev-env` project in Sentry.io
+  # shellcheck disable=SC2155
+  export SENTRY_LAST_EVENT=$(
+    SENTRY_DSN=https://23670f54c6254bfd9b7de106637808e9@o1.ingest.sentry.io/1492057 \
```

Comment:
Can we either 1) have our bootstrap script inject this into user profile startup scripts or 2) allow opt out?

Reply:
We can export it inside of `.envrc`, thus, it becomes an env variable accessible to any of our make targets.
We could have folks opt-out by having a non-tracked file that `.envrc` sources at the end.

Side note, as an opt-in mechanism we can have folks define env variables in a `.env` file (using `dotenv-cli` and https://github.com/getsentry/sentry/blob/9e4be2b9a919cfb974fdfda610dd6c666bfbec78/.envrc#L181-L184).